src/main/java/seedu/address/model/lesson/Lesson.java

```diff
@@ -126,38 +131,26 @@ public boolean equals(Object other) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(date, timeRange, subject, homework);
-    }
-
-    @Override
-    public String toString() {
-        final StringBuilder builder = new StringBuilder();
-        String typeOfLesson = isRecurring() ? RECURRING : MAKEUP;
-        builder.append(typeOfLesson)
-            .append("\n")
-            .append(getDate())
-            .append("\nTime: ")
-            .append(getTimeRange())
-            .append("\nSubject: ")
-            .append(getSubject());
-
-        Set<Homework> homework = getHomework();
-        if (!homework.isEmpty()) {
-            builder.append("\nHomework: ");
-            homework.forEach(builder::append);
-        }
-        return builder.toString();
+        // use this method for custom fields hashing instead of implementing your own
+        return Objects.hash(startDate, timeRange, subject, homework);
     }
 
     /**
-     * Compares this Lesson object with the other Lesson object.
+     * Compares this RecurringLesson object with the other Lesson object.
      *
      * @param other The Lesson object to compare with.
      * @return 1, if this is later than other;0 if equal; -1 if this is earlier.
      */
     @Override
     public int compareTo(Lesson other) {
-        int compareDate = getDate().compareTo(other.getDate());
+        /*
+        Date represents the date of this lesson that is relevant to the comparison.
+        i.e. The upcoming date for recurring lessons; or the start date for makeup lessons
+        (since makeup lesson is a one-time event).
+         */
+        Date date = isRecurring() ? getNextDate() : getStartDate();
+        Date dateToCompare = other.isRecurring() ? other.getNextDate() : other.getStartDate();
+        int compareDate = date.compareTo(dateToCompare);
```

Comment:
Actually, I'm not sure because if we do this, we can remove the first check below but we end up with the exact same method for both the compareTo() method in RecurringLesson and MakeUpLesson classes since we have to check if the other lesson is recurring to allow comparison between Recurring and Makeup lessons?
```
        Date date = isRecurring() ? getNextDate() : getStartDate();
```